### PR TITLE
Implement scheduled irigation

### PR DESCRIPTION
**Pull request by luciiii6**

### CODE REVIEW

File: is_iot_sink/irrigation/scheduled/scheduled_irrigation.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        self.running = False
+        self.global_thread.join()
+
+    def __run(self):
+
+        #to do
+        while self.running:
+            appointment = self.__mongo_client.read_first_appointment()
+
+            if appointment is not None:
```
Comment (DariusCr) [inline is_iot_sink/irrigation/scheduled/scheduled_irrigation.py]: please use if appointment

File: is_iot_sink/irrigation/scheduled/scheduled_irrigation.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                delay = self.__get_delay(appointment['timestamp'])
+                duration = appointment['duration']
+                self.__valve_manager.start_valves_cycle(duration, delay)
+
```
Comment (DariusCr) [inline is_iot_sink/irrigation/scheduled/scheduled_irrigation.py:43]: why so many blank lines?

Comment (luciiii6) [reply is_iot_sink/irrigation/scheduled/scheduled_irrigation.py:43]: this is not done, it has a comment as #to do

File: is_iot_sink/mqtt/mqtt_client_base.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import queue
+import os
+import paho.mqtt.client as mqtt
+from is_iot_sink.logger import LOG
+from is_iot_sink.settings import Settings
+
+class MQTTClientBase:
+    def __init__(self, settings: Settings, name = ""):
+        self._settings = settings
```
Comment (DariusCr) [inline is_iot_sink/mqtt/mqtt_client_base.py:9]: we should decide on a convention and respect it regarding to using 1 or 2 underscores for private marked class members

File: is_iot_sink/irrigation/scheduled/scheduled_irrigation.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import threading
+
+
+class ScheduledIrrigation:
+    def __init__(self, valve_manager: ValveManager, mongo_client: MongoClient):
+        super().__init__()
+        self.__valve_manager = valve_manager
+        self.__mongo_client = mongo_client
+        self.mode = IrrigationMode.SCHEDULED
+        self.running = False
```
Comment (DariusCr) [inline is_iot_sink/irrigation/scheduled/scheduled_irrigation.py:18]: why are these 2 not marked as private as well, as they don't seem to be modified or used outside the class

File: is_iot_sink/mqtt/mqtt_client_base.py (hunk 3 above)
Comment (pdany1116) [reply is_iot_sink/mqtt/mqtt_client_base.py:9]: In python, double underscore `__` means private and single underscore `_` means protected.

File: is_iot_sink/irrigation/scheduled/scheduled_irrigation.py (hunk 4 above)
Comment (pdany1116) [reply is_iot_sink/irrigation/scheduled/scheduled_irrigation.py:18]: Good point, missed them. 

The irrigation mode should have a getter. We want the Sink module to return the actual irrgation mode.
https://github.com/luciiii6/is-iot-sink/pull/3/files#diff-316c0baad536662e1332877cec5b8aff4cc321f81f82a67ae6019e88d5c28872R44

**Pull request merged**